pkgs/tools/filesystems/ceph/default.nix

```diff
@@ -182,6 +196,8 @@ in rec {
       "-DWITH_TESTS=OFF"
       # TODO breaks with sandbox, tries to download stuff with npm
       "-DWITH_MGR_DASHBOARD_FRONTEND=OFF"
+      # WITH_XFS has been set default ON from Ceph 16, keeping it optional in nixpkgs for now
+      ''-DWITH_XFS=${if optLibxfs != null then "ON" else "OFF"}''
```

Comment:
Overriding packages with null is an anti pattern and new occurrences of it are strongly discouraged.

Reply:
Bump

Reply:
This isn't really a new occurrence of this pattern; it's fixing the existing one.

That said, it's definitely worth revisiting if we want to keep this optional; maybe we just want to require this on !Darwin.

Reply:
Agree with @lukegb here. I wan't really planning to fixup the package inputs in this PR. My only goal here is to ensure that the existing behavior is preserved while upgrading. A general cleaup of the expression could be nice, but I hope that won't hold back the bump of Ceph versions here.

Reply:
I'm not sure how this is an instance of "Overriding packages with null". 

`optLibxfs` is defined above as

```
let
  shouldUsePkg = pkg: if pkg != null && pkg.meta.available then pkg else null;
  # ...
  optLibxfs = shouldUsePkg libxfs;
```

`libxfs` is defined in `all-packages.nix` as:
```
  xfsprogs = callPackage ../tools/filesystems/xfsprogs { };
  libxfs = xfsprogs.dev;
```

`xfsprogs/default.nix` in turn has:

```
stdenv.mkDerivation rec {
#...
  outputs = [ "bin" "dev" "out" "doc" ];
# ...
  meta = with lib; {
#...
    platforms = platforms.linux;
```

I don't have a non-Linux system to test on, but I'm guessing `libxfs.meta.available` would be false on non-Linux rather than `libxfs` being null.

I don't see any relevant overriding happening.

@SuperSandro2000 Could you elaborate further on what the anti-pattern is? (especially if you consider this a blocker) Is defining a separate variable that can be `null` a problem?

Reply:
In general, I think the view is that we'd prefer that packages which are in the argumentset of a package are not defaulted to null. This is because it becomes much harder to notice when changes (e.g. renames, alias changes) break downstream packages, because the downstream packages continue to compile but silently omit functionality that used to be there.

This pattern has been used in the past to allow for disabling features by nulling out the dependencies; in general, we'd prefer a separate boolean flag (e.g. named `withLibxfs` or `withXfsSupport`) which then changes whether or not the package is actually used (e.g. in buildInputs).

withLibxfs can, for instance, then be defaulted to stdenv.isLinux.